New packkage: games-puzzle/blockout

abecerra opened the pull request:
Signed-off-by: Andrés Becerra <andres.becerra@gmail.com>
Tested-by: Andrés Becerra <andres.becerra@gmail.com>
Bug: https://bugs.gentoo.org/545604
Closes: https://bugs.gentoo.org/545604

- New ebuild authored by Vladimir <vl409@yandex.ru>
- Modifications to the original ebuild and pull request creation by Andrés Becerra who could proxy-maintain it.

abecerra commented:
> ## Pull Request assignment
> 
> _Submitter_: @abecerra _Areas affected_: ebuilds _Packages affected_: games-puzzle/blockout
> 
> **games-puzzle/blockout**: @gentoo/proxy-maint (new package)
> ## Linked bugs
> 
> Bugs linked: [545604](https://bugs.gentoo.org/545604)
> ## New packages
> 
> This Pull Request appears to be introducing new packages only. Due to limited manpower, adding new packages is considered low priority. This does not mean that your Pull Request will not receive any attention, however, it might take quite some time for it to be reviewed. 

In defense of adding this little game, the source code has not changed since 2014-05 and it would imply little maintenance in the future.

ceamac commented on games-puzzle/blockout/blockout-2.5.ebuild:
```diff
@@ -0,0 +1,53 @@
+src_unpack() {
+	unpack ${A}
+	mv "BL_SRC" ${P}
+}
```
No need for a `src_unpack` function, define `S="${WORKDIR}"/BL_SRC` instead

abecerra replied:
Removed

ceamac commented on games-puzzle/blockout/blockout-2.5.ebuild:
```diff
@@ -0,0 +1,53 @@
+src_prepare() {
+	eapply "${FILESDIR}"/${P}-datadir.patch
+	eapply_user
+}
```
You don't need `src_prepare` either, define a `PATCHES` variable and the default implementation will do it

abecerra replied:
Removed

ceamac commented on games-puzzle/blockout/blockout-2.5.ebuild:
```diff
@@ -0,0 +1,53 @@
+	eapply_user
+}
+
+src_install() {
```
Compilation should not take place in the install phase

abecerra replied:
Moved

ceamac reviewed (commented):
I took a quick look and there are a few issues with this package.
```
 * VDB: detected possibly incorrect RDEPEND (games-puzzle/blockout-2.5)
 *                  > media-libs/alsa-lib
 * x11-libs/libXext < 
 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/blockout
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/blockout
```

I suggest you fix as many issues as you can and consider adding this to GURU instead until it matures.

Thank you!

abecerra commented:
> I took a quick look and there are a few issues with this package.
> 
> ```
>  * VDB: detected possibly incorrect RDEPEND (games-puzzle/blockout-2.5)
>  *                  > media-libs/alsa-lib
>  * x11-libs/libXext < 
>  * QA Notice: Files built without respecting CFLAGS have been detected
>  *  Please include the following list of files in your report:
>  * /usr/bin/blockout
>  * QA Notice: Files built without respecting LDFLAGS have been detected
>  *  Please include the following list of files in your report:
>  * /usr/bin/blockout
> ```
> 
> I suggest you fix as many issues as you can and consider adding this to GURU instead until it matures.
> 
> Thank you!

Removed libXext and added alsa-lib
